### PR TITLE
add map projection expression

### PR DESCRIPTION
**Pull request by jievince**
https://github.com/vesoft-inc/nebula-graph/pull/1292 and https://github.com/vesoft-inc/nebula-storage/pull/537
depends on this pr

### CODE REVIEW

File: src/common/expression/ContainerExpression.cpp [hunk 1]
```diff
@@ -236,4 +238,109 @@ void MapExpression::accept(ExprVisitor *visitor) {
...
+}
+
+
+void MapProjectionExpression::writeTo(Encoder &encoder) const {
+    encoder << kind();
+    encoder << mapVarName_;
+    encoder << size();
+    for (auto &kv : items_) {
+        encoder << kv.first;
+        bool v_exist = (kv.second != nullptr);
```
Comment (CPWstatic) [inline src/common/expression/ContainerExpression.cpp:317]: v_exist -> vExist

File: src/common/expression/ContainerExpression.cpp [hunk 2]
```diff
@@ -236,4 +238,109 @@ void MapExpression::accept(ExprVisitor *visitor) {
...
+}
+
+
+void MapProjectionExpression::resetFrom(Decoder &decoder) {
+    mapVarName_ = decoder.readStr();
+    auto size = decoder.readSize();
+    items_.reserve(size);
+    for (auto i = 0u; i < size; i++) {
+        auto str = decoder.readStr();
+        bool v_exist = decoder.readValue().getBool();
```
Comment (CPWstatic) [inline src/common/expression/ContainerExpression.cpp:333]: v_exist -> vExist

File: src/common/expression/ContainerExpression.cpp [hunk 3]
```diff
@@ -236,4 +238,109 @@ void MapExpression::accept(ExprVisitor *visitor) {
     visitor->visit(this);
 }
 
+
+std::string MapProjectionExpression::toString() const {
+    return mapVarName_;
```
Comment (CPWstatic) [inline src/common/expression/ContainerExpression.cpp:243]: Why only returns mapVarName_

**Comment by jievince** [comment]
Should migrate to nebula repo

**Pull request closed without merging**